crates/matrix-sdk-crypto/src/machine.rs

```diff
@@ -856,6 +856,114 @@ impl OlmMachine {
         }
     }
 
+    fn record_message_id(event: &Raw<AnyToDeviceEvent>) {
+        use serde::Deserialize;
+
+        #[derive(Deserialize)]
+        struct ContentStub<'a> {
+            #[serde(borrow, rename = "org.matrix.msgid")]
+            message_id: JsOption<&'a str>,
+        }
+
+        #[derive(Deserialize)]
+        struct ToDeviceStub<'a> {
+            #[serde(borrow)]
+            content: ContentStub<'a>,
+        }
+
+        let message_id = event
+            .deserialize_as::<ToDeviceStub<'_>>()
+            .ok()
+            .and_then(|t| t.content.message_id.into_option());
+
+        tracing::Span::current().record("message_id", message_id);
```

Comment:
Do we want to record a `None` here?

Reply:
I would say no, it's going to be far too often a `None`.

Reply:
Then let's add an `if let` around this.